sfcasts/ep1/computed.md

```diff
@@ -1,228 +1,192 @@
 
 Excellent!
 
 ## How computed works
- 
-So here's the deal. As soon as you have a key under the `computed` object, this
-becomes available in the template as a variable. In fact, `vue` will create a
-variable with a `getter` set to our method and assign it to its magical `this` keyword.
-But most importantly, if *anything* inside that method changes, weather it be a
-`data` property or a `prop`, `Vue` will automatically re-run that method for us
-and recalculate the rendered DOM in a *very* optimized way. *Wow!* Your world
-just changed forever!
-
-So we'll copy `componentClass` and up here in the template, very simply we'll say
-`:class="componentClass"`. I can also organize this better and move our `<div>`
-on a single line. That's it, ESlint, don't be mad at me!
- 
-So up until now, we know that `vue` adds any `data` `props` or `methods` to the vue
-instance, which means that we can reference those inside of our template.
-Well, `computed` is the fourth and final thing that gets added to that instance
-and as we just said, `Vue` adds any key under `computed` as a `property` with its
-`getter` set to that method.
-
-So in up here back in the template, we can just reference `componentClass`,
-which is really `this.componentClass`. And behind the scenes `vue` is going to
-call this method `componentClass` to get that and update it whenever necessary.
-
-Because this is being added to the `vue` instance, just like `methods`, `data` or
-`props`, it means that we can actually reference it with the `this` variable.
-Just to prove it, inside of our `toggleCollapsed` method, let's say
-`console.log(this.componentClass())`.
- 
- 
- 
-Notice that PHPStorm tries to auto complete that with parentheses, but that's not
-correct! We need to reference it like a property, even though behind the scenes,
-we know view is going to call this method.
+
+So here's the deal. As soon as you have a key under the `computed` option, it
+becomes available in the template as a *variable*.
+
+Copy `componentClass` and, up in the template, very simply, we'll say
+`:class="componentClass"`.
+
+So up until now, we know that Vue adds any `data`, `props` or `methods` to the Vue
+instance, which means that we can reference those inside our template.
+Well, `computed` is the fourth and *final* thing that gets added to the instance.
+`Vue` adds each key under `computed` as a *property*.
+
+This means that, up in the template, we can just reference `componentClass`,
+which is really `this.componentClass`. But behind the scenes, when we access that
+property, Vue will *actually* call the `componentClass` method to get it. It's able
+to do that thanks to the fake getter properties we saw earlier.
+
+And really, the only difference between `methods` and `computed` is the syntax: we
+use methods like methods and computed like properties. Oh, and Vue caches computed
+properties so that it only needs to call our function when something actually
+*changes*.
```

Comment:
This sort of tries to explain what I said in my last comment, but somehow it isn't resonating with me... May be say:

```
In reality, a `computed` *function* will create a *property* of the same name with some magic in it! Every time something changes *inside* that function, it will get executed for us, updating the property and re-rendering the template! [Add some exclamation here]
```

Reply:
I think we're thinking about the reactivity in a slightly different way. I'm not sure if both are right or just one of us.

I think of:

> When you render, Vue tracks all props/data that were referenced ANYWHERE. And later, if any of these change, the ENTIRE template is re-rendered (of course, Vue intelligently updates the DOM)

You are thinking:

> When you render, if you use a computed property and that references a `name` data, then the `name` data changes, Vue will re-execute *just* your computed property function.

Those might just be two different ways to thinking about reactivity that are both correct. I'm not sure.

Reply:
Right, to be optimized, Vue tracks everything you call inside a computed property. If you change it, it will re-call that method to update the return value. When that value updates, Vue can render *just* the parts of the template where that value is relevant. It's a really cool thing, it deviates a bit from what you say at the beginning.

Reply:
https://vuejs.org/v2/guide/computed.html